File: ios/objc/StorylyDemo/StorylyDemo/CustomThemeViewController.m
//
//  UIViewController+CustomThemeViewController.m
//  StorylyDemo
//
//  Created by Haldun Melih Fadillioglu on 3.02.2021.
//  Copyright © 2021 App Samurai Inc. All rights reserved.
//

#import "CustomThemeViewController.h"

@interface CustomThemeViewContoller ()

@property (weak, nonatomic) IBOutlet StorylyView *storylyViewDefaultTheme;
@property (weak, nonatomic) IBOutlet StorylyView *storylyViewCustomTheme;

@end

@implementation CustomThemeViewContoller

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSString *YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD = @"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjQ3LCJhcHBfaWQiOjEyODcsImluc19pZCI6MTYzMX0.mjY5nZ5mPk0H6v5xiexJnpNzGGsn335Cm2mYlzCWo4A";
    
    self.storylyViewDefaultTheme.storylyInit = [[StorylyInit alloc] initWithStorylyId: YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD];
    self.storylyViewDefaultTheme.rootViewController = self;

    
    self.storylyViewCustomTheme.storylyInit = [[StorylyInit alloc] initWithStorylyId: YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD];
    self.storylyViewCustomTheme.rootViewController = self;
    
//    self.storylyViewCustomTheme.storyGroupTextColor = [UIColor colorWithRed: 240.0/255.0 green: 57.0/255.0 blue: 50.0/255.0 alpha: 1.0];
//    self.storylyViewCustomTheme.storyGroupIconBackgroundColor = [UIColor colorWithRed: 76.0/25.0 green: 175.0/255.0 blue: 80.0/255.0 alpha: 1.0];
    
    self.storylyViewCustomTheme.storyGroupIconBorderColorSeen = [[NSArray alloc] initWithObjects:
                                                                 [UIColor colorWithRed: 197.0/255.0 green: 172.0/255.0 blue: 165.0/255.0 alpha: 1.0],
                                                                 [UIColor colorWithRed: 1.0 green: 1.0 blue: 1.0 alpha: 1.0], nil];
    
    self.storylyViewCustomTheme.storyGroupIconBorderColorNotSeen = [[NSArray alloc] initWithObjects:
                                                                    [UIColor colorWithRed: 251.0/255.0 green: 50.0/255.0 blue: 0 alpha: 1.0],
                                                                    [UIColor colorWithRed: 1.0 green: 235.0/255.0 blue: 59.0/255.0 alpha: 1.0], nil];
    
//    self.storylyViewCustomTheme.storyGroupPinIconColor = [UIColor colorWithRed: 63.0/255.0 green: 81.0/255.0 blue: 181.0/255.0 alpha: 1.0];

//    self.storylyViewCustomTheme.storyItemTextColor = [UIColor colorWithRed: 255.0/255.0 green: 0 blue: 87.0/255.0 alpha: 1.0];

    self.storylyViewCustomTheme.storyItemIconBorderColor = [[NSArray alloc] initWithObjects:
                                                           [UIColor colorWithRed: 251.0/255.0 green: 50.0/255.0 blue: 0 alpha: 1.0],
                                                           [UIColor colorWithRed: 255.0/255.0 green: 235.0/255.0 blue: 59.0/255.0 alpha: 1.0], nil];
    
    self.storylyViewCustomTheme.storylyItemProgressBarColor =  [[NSArray alloc] initWithObjects:
                                                                [UIColor colorWithRed: 251.0/255.0 green: 50.0/255.0 blue: 0 alpha: 1.0],
                                                                [UIColor colorWithRed: 255.0/255.0 green: 235.0/255.0 blue: 59.0/255.0 alpha: 1.0], nil];

}

@end

File: ios/swift/StorylyDemo/StorylyDemo/CustomThemeViewController.swift
//
//  CustomThemeViewController.swift
//  StorylyDemo
//
//  Created by Haldun  Fadillioglu on 2.02.2021.
//  Copyright © 2021 App Samurai Inc. All rights reserved.
//

import UIKit
import Storyly

class CustomThemeViewController: UIViewController {

    @IBOutlet weak var storylyViewDefaultTheme: StorylyView!
    @IBOutlet weak var storylyViewCustomTheme: StorylyView!

    override func viewDidLoad() {
        super.viewDidLoad()

        storylyViewDefaultTheme.storylyInit = StorylyInit(storylyId: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40")
        storylyViewDefaultTheme.rootViewController = self
        
        storylyViewCustomTheme.storylyInit = StorylyInit(storylyId: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40")
        storylyViewCustomTheme.rootViewController = self
        
        storylyViewCustomTheme.storyGroupTextColor = UIColor(red: 240/255, green: 57/255, blue: 50/255, alpha: 1.0)

        storylyViewCustomTheme.storyGroupIconBackgroundColor = UIColor(red: 76/255, green: 175/255, blue: 80/255, alpha: 1.0)

        let seenBorder = [UIColor(red: 197/255, green: 172/255, blue: 165/255, alpha: 1.0), UIColor.black]
        storylyViewCustomTheme.storyGroupIconBorderColorSeen = seenBorder
        
        let notSeenBorder = [UIColor(red: 251/255, green: 50/255, blue: 0, alpha: 1.0),
                            UIColor(red: 255/255, green: 235/255, blue: 59/255, alpha: 1.0)]
        storylyViewCustomTheme.storyGroupIconBorderColorNotSeen = notSeenBorder
//
//        storylyViewCustomTheme.storyGroupPinIconColor = UIColor(red: 63/255, green: 81/255, blue: 181/255, alpha: 1.0)
//
//        storylyViewCustomTheme.storyItemTextColor = UIColor(red: 255/255, green: 0, blue: 87/255, alpha: 1.0)
//
        let iconBorderColor = [UIColor(red: 251/255, green: 50/255, blue: 0, alpha: 1.0),
                               UIColor(red: 255/255, green: 235/255, blue: 59/255, alpha: 1.0)]
        storylyViewCustomTheme.storyItemIconBorderColor = iconBorderColor
//        
        let progressbarColor = [UIColor(red: 251/255, green: 50/255, blue: 0, alpha: 1.0),
                                UIColor(red: 255/255, green: 235/255, blue: 59/255, alpha: 1.0)]
        storylyViewCustomTheme.storylyItemProgressBarColor = progressbarColor
    }
}
